apps/server/src/modules/task-card/uc/task-card.uc.ts

```diff
@@ -109,6 +135,18 @@ export class TaskCardUc {
 			cardElements.push(...texts);
 		}
 
+		if (params.visibleAtDate) {
+			card.visibleAtDate = params.visibleAtDate;
+		}
+
+		if (params.dueDate) {
+			card.dueDate = params.dueDate;
+		}
+
+		if (!card.isVisibleBeforeDueDate()) {
+			throw new BadRequestException();
```

Comment:
not sure if this is a bad request or a validation exception
also, the fact we throw it here and not in the mapping, as we do for example with the title, is correct, because of business logic for default dates

Reply:
which kind of exception do you recommend to use? as there is no "validationException"

Reply:
import { ValidationError } from '@shared/common';